app/models/manageiq/providers/workflows/automation_manager/workflow_instance.rb

```diff
@@ -49,13 +49,15 @@ def queue_callback(state, message, _result)
   end
 
   def run(args = {})
-    zone, role, object_type, object_id = args.values_at(:zone, :role, :object_type, :object_id)
+    queue_args = args.slice(:zone, :role, :object_type, :object_id)
+    zone, role, object_type, object_id = queue_args.values_at(:zone, :role, :object_type, :object_id)
```

Comment:
nit: think you may be able to get away with keeping `s/queue_args/args/` (one less diff line)

Reply:
I tried this originally but miq_queue automatically tacks on a miq_task_id to the args on delivery and that ended up throwing an exception because it wasn't a recognized keyword for run_queue

Reply:
Good catch. I really don't like that `$miq_task_id` logic